### ISSUE TITLE
Auto-Crating Simulation issue with Storage Buses

### ISSUE THREAD

**Issue by vgamesx1** [AppliedEnergistics/Applied-Energistics-2#1036]
Hello I just wanted you to be aware that this is still an issue, I'm running the latest Beta rv2-11 and I'm having issues similar to #682 where having a storage bus attached to a barrel causes simulation hang ups, although the difference is that I don't have multiple barrels, just a single cobblestone barrel and it will lock up any crafting recipe at random.

I know other people have already commented about this, but it has plagued my world for quite a long time now, I just recently figured it out because I went nuts smashing a bunch of cables trying to find the cause.

**Comment by thatsIch** [comment]
what kind of barrel is it? is it JABBA?

**Comment by vgamesx1** [comment]
Oh yeah sorry about that, it was a JABBA barrel, one with a few storage upgrades as well.

**Comment by yueh** [comment]
Is the barrel in the new setup also locked? The angle is pretty bad, but I would assume it is not.

**Comment by vgamesx1** [comment]
The bottom isn't so it isn't an exact copy but it was still causing that issue, so I don't know if it has to do with the transfer node specifically but the back one piping out to a appears to be what makes this become an issue, is this possibly because its constantly pulling one item out?

**Comment by Prodygy** [comment]
I recently encountered this same issue (or what appears to be the same issue) in a private multiplayer world. I had an AE storage bus connected to a Deep Storage Unit. There is also an Item Transfer Node connected to the DSU which pulls out cobble to put into a Matter Condenser for making Singularities:

![2015-05-03_21 27 36](https://cloud.githubusercontent.com/assets/1151639/7448906/0a0ac242-f1dc-11e4-9f81-1a7d8b387b25.png)

When everything was attached, I was having difficulty doing any crafting recipes. Autocrafting was partially working (I could only craft one item at a time) but it gradually grew worse until it completely stopped.

When I removed the Transfer Node connected to the DSU, autocrafting started working again:

![2015-05-03_21 27 44](https://cloud.githubusercontent.com/assets/1151639/7448907/11fec11a-f1dc-11e4-953a-7d4fb95e23e7.png)

As vgamesx1 alluded, my guess is that the Transfer Node was somehow preventing the storage bus from accessing the DSU because of the 64 item limit per transfer. This would make sense since JABBA barrels behave similarly. I still find it interesting that none of the autocrafting was working though.

**Comment by thatsIch** [comment]
my guess is, that you update the DSUs constantly, which reflects the storage bus refreshing its cache and thus halting the content collection.

guess need to install XU to test it now.

What configuration are you using in the transfer node?
Might be able to replicate that.

**Comment by Prodygy** [comment]
Originally there were 2 stacks of speed upgrades and 4 stacks of world interaction upgrades. I was able to reproduce this behavior with 1 stack each, but it was much less consistent -- sometimes I was able to craft, other times not. I did notice that it was easier to reproduce when crafting large quantities (100s or 1000s).

In my testing, the DSU contained cobble, but it was preventing crafting of items that didn't contain cobble, such as sticks.

**Comment by Lothmorg** [comment]
I have been having a similar problem. I think I have tracked it down somewhat. If I have an ender quarry connected via tesseract/item conduit to a DSU (with a storage bus), then I can no longer craft anything. I believe the DSU is updating too fast as stated above and ties up the ME network. As soon as I break the tesseract/DSU connection, or remove the storage bus, everything works.

**Comment by m0rgs** [comment]
I wanted to comment and add that this issue occurs with the Black Hole Storage from **SuperMassiveTech**.

If I put a storage bus on the Black Hole Storage, autocrafting of items "start" remains grey about 50% of the time.

**Comment by yueh** [comment]
The main issue is still, that none of us is able to replicate the issue.

I tried probably a dozen of different setups including some large setups with a bunch of cobble gens storing items directly into a DSU. But without any issue.

**Comment by m0rgs** [comment]
I just created a world in creative with minimal mods and default configs (deleted config folder before startup) that is susceptible to this bug.

BHS = Black Hole Storage (2^40 storage of a single item)
SMT = Super Massive Tech

Set up some big storage (i.e. BHS from SMT) which is having max world interaction upgrades of cobble pumped into it. I set up 6 item transfer nodes, each with 6 stacks of World Interaction and 1 stack of speed upgrade, attached to a cobblestone block adjacent to lava & water for cobble gen. It was generating about 1500 cobble/s, give or take. I also set the BHS  to priority 1000.

The cobble was being inserted directly into the BHS, NOT an interface. 

Next, set up a recipe that isn't related to the cobble, in my case I set up a recipe for crafting table. Throw some wood into your storage cells.

Open your terminal and instruct it to craft a crafting table. It works 20-50% of the time. The other times, "start" remains greyed out.

http://imgur.com/a/UZDC7

**Comment by XFactHD** [comment]
As far as I know from a different issue the Black Hole Storage doesn't implement the DSU interface.

**Comment by vgamesx1** [comment]
@yueh not sure why you're having trouble recreating the issue, because I was able to replicate it on a bare bones install having only

appliedenergistics2-rv2-stable-3.jar
CodeChickenCore-1.7.10-1.0.6.43-universal.jar
extrautilities-1.2.5.jar
NotEnoughItems-1.7.10-1.0.4.107-universal.jar
Jabba-1.2.1a_1.7.10.jar

Here's how:
![2015-06-01_13 10 31](https://cloud.githubusercontent.com/assets/11490777/7918628/c51a8fc8-0860-11e5-83aa-945e1d3aec55.png)
Fully upgraded jabba barrel being feed by a fully upgraded transfer node
![2015-06-01_13 11 38](https://cloud.githubusercontent.com/assets/11490777/7918654/0710f1ce-0861-11e5-9ea6-9a8aa1e9818d.png)
that alone however, will only cause minor irritation as the issue does occur but rarely in this state, to fully break it you'll need something else acting on the barrel such as this (I added a few speed upgrades by the way)
![2015-06-01_13 10 47](https://cloud.githubusercontent.com/assets/11490777/7918707/64f00014-0861-11e5-80cd-5dc1db0edc38.png)
now trying to craft wood to planks then into chests will cause this to happen some what regularly.
![2015-06-01_13 10 11](https://cloud.githubusercontent.com/assets/11490777/7918782/d15f197e-0861-11e5-8d8b-57187d950fc2.png)

![2015-06-01_13 09 59](https://cloud.githubusercontent.com/assets/11490777/7918773/bd4c83a4-0861-11e5-83b5-0f49e5f59806.png)

**Comment by m0rgs** [comment]
> As far as I know from a different issue the Black Hole Storage doesn't implement the DSU interface.

So?

I installed MFR on my test setup, replaced the BHS for a DSU and the issue persists.

**Comment by Lothmorg** [comment]
I have just recreated the issue 100% with a simple setup and minimal mods.
![2015-06-01_18 00 02](https://cloud.githubusercontent.com/assets/12576134/7925303/bd47fd94-0888-11e5-92ec-2a150db8fb18.png)
This shows a basic setup being fed by an active Ender Quarry (speed 3) via Tesseract. The items come from the quarry (via Ender-IO Item Conduit), insert cobble into DSU (Priority 1), the rest into the chest (Priority 0) and then are imported into the system. The DSU has a Storage Bus on it.
![2015-06-01_18 00 21](https://cloud.githubusercontent.com/assets/12576134/7925360/42cb3e86-0889-11e5-8055-5689a089c717.png)
This is a simple Ender Quarry with Speed 3 upgrade on it, and of course the other end of the Tesseract.

![2015-06-01_18 00 41](https://cloud.githubusercontent.com/assets/12576134/7925374/6c1cda6a-0889-11e5-93ac-e0739ad9543d.png)
Before I activate the Ender Quarry, I can craft just fine.

![2015-06-01_18 00 53](https://cloud.githubusercontent.com/assets/12576134/7925386/7d9e67d6-0889-11e5-86c4-7e34e9b7e3db.png)
Once I start the Ender Quarry (give it power), then the DSU starts filling - fast, and I can no longer craft.
Removing the Storage Bus, or deactivating the Ender Quarry resumes normal crafting function.
These are the mods I have used.
appliedenergistics2-rv2-stable-3
CoFHCore-[1.7.10]3.0.2-262
CoFHLib-[1.7.10]1.0.1-151
EnderIO-1.7.10-2.2.8.381
extrautilities-1.2.5
MineFactoryReloaded-[1.7.10]2.8.0-104
ThermalDynamics-[1.7.10]1.0.0-122
ThermalExpansion-[1.7.10]4.0.1-182
ThermalFoundation-[1.7.10]1.0.0-81

I know it's frustrating to hear of a bug when you've poured countless hours into something, but we are just trying to help you make this better. Thanks for all you do!

**Comment by yueh** [comment]
These are not a minimal mod lists, especially things like FastCraft should never be used to reproduce issues. Basically anything besides AE2, something providing an external storage like MFR or Jabba (vanilla chests would be even better) and maybe XU is the maximum possible setup as not all mods are availabe for indev or are just making it way harder to reproduce (like introducing their own bugs).

Also we need to debug these issues, but the actual bug occurs way before the crafting window appears.
Spending a couple of minutes each time and just fail every other attempt is just not feasible. With some bad luck we can easily spend 30 minutes without any hint, but a single misclick can easily destroy the current attempt and waste another 30 minutes or even more. Not even speaking about the amount of data to track and only being able to deduce anything from it after it actually failed.
Basically we cannot depend on hoping it fails sometimes, it needs to fail with every attempt (or at least some very high chance, probably over 95%).

**Comment by Lothmorg** [comment]
My setup above fails 100% of the time. If the DSU is updating too fast, the storage bus on it causes the assemblers to hang.

**Comment by m0rgs** [comment]
For the sake of curiosity I have re-created the bug using these two mods ONLY, no MFR, no fastcraft, no invtweaks/mousetweaks/WAILA/NEI etc:

* appliedenergistics2-rv2-stable-3.jar
* extrautilities-1.2.5.jar

Extra Utilities is the mod of choice to moving the amount of items fast enough to reproduce this bug, but there are other mods that could do it. I suspect it is just getting bugged trying to update its item count too often which locks the network up.

This setup fails about 80% of the time for me.

http://imgur.com/a/LkQjv

**Comment by XFactHD** [comment]
@m0rgs the issue also stated that the me system could only see a stack of the stored item from the BHS. Could have changed since the other issue.

Another question I have: Could someone please explain what the point is of having a tesseract put the cobblestone directly into the DSU and the rest into the me system and not just everything through the me system. That's what I do and with about 200 mods I don't have this problem.
